### PR TITLE
[FIX] mail: mark all needaction as read when opening the thread

### PR DESCRIPTION
**Pull request by phenix-factory**
This was the case before the refactoring.

task-2411114

### CODE REVIEW

File: addons/mail/static/src/components/chatter/chatter.js [hunk 1]
```diff
@@ -60,6 +60,10 @@ class Chatter extends Component {
         return this.env.models['mail.chatter'].get(this.props.chatterLocalId);
     }
 
+    mounted() {
+        this.chatter.thread.markNeedactionMessagesAsRead();
+    }
```
Comment (seb-odoo) [inline addons/mail/static/src/components/chatter/chatter.js:65]: Prefer fix in models instead, `_onChangeThreadViews` is already doing similar stuff.

Also make sure RPC is not done if there's no needaction.

Comment (phenix-factory) [reply addons/mail/static/src/components/chatter/chatter.js]: This is clearing ALL the notifications !
My guess is that `_onChangeThreadViews` is computed even when you don't open the thread

Comment (seb-odoo) [reply addons/mail/static/src/components/chatter/chatter.js]: Yes, all computes are always called at create, that's why there is a condition on threadView > 0

Comment (seb-odoo) [reply addons/mail/static/src/components/chatter/chatter.js]: I think you need to exclude mailbox from this. But please return to the compute version, not mounted.

Actually you need to exclude all non-origin thread, as per done in https://github.com/odoo/odoo/pull/62981

File: addons/mail/static/src/models/thread/thread.js [hunk 2]
```diff
@@ -1077,6 +1077,16 @@ function factory(dependencies) {
 
         /**
          * @private
+         * @returns {boolean}
+         */
+        _computeHasNeedaction() {
+            // TODO: this could rely on the message_needaction_counter when fixed
+            return this.needactionMessages.length > 0;
+        }
```
Comment (seb-odoo) [inline addons/mail/static/src/models/thread/thread.js:1085]: don't need this compute, just directly use this.needactionMessagesAsOriginThread.length in the other compute

File: addons/mail/static/src/models/thread/thread.js [hunk 3]
```diff
@@ -1509,23 +1510,23 @@ function factory(dependencies) {
...
-                return;
+            /**
+             * Fetches followers of chats when they are displayed for the first
+             * time. This is necessary to clean the followers.
+             * @see `_onChangeFollowersPartner` for more information.
+             */
+            if (this.channel_type === 'chat' && !this.areFollowersLoaded) {
+                this.refreshFollowers();
+            }
+            if (this.needactionMessagesAsOriginThread.length) {
```
Comment (seb-odoo) [inline addons/mail/static/src/models/thread/thread.js:1527]: We like to make integer compare explicit (so > 0 in this case)

Comment (alexkuhn) [reply addons/mail/static/src/models/thread/thread.js:1527]: Agree, truthy/falsy is too subjective when it's not boolean (that's why it's different from language to language...), so it's best using actual boolean value for conditionals.

**Review by seb-odoo — APPROVED** [review]
Looks good, @alexkuhn final check?

File: addons/mail/static/src/models/thread/thread.js [hunk 4]
```diff
@@ -1509,23 +1510,23 @@ function factory(dependencies) {
...
-            if (this.areFollowersLoaded) {
-                return;
+            /**
+             * Fetches followers of chats when they are displayed for the first
+             * time. This is necessary to clean the followers.
+             * @see `_onChangeFollowersPartner` for more information.
+             */
+            if (this.channel_type === 'chat' && !this.areFollowersLoaded) {
+                this.refreshFollowers();
+            }
```
Comment (alexkuhn) [inline addons/mail/static/src/models/thread/thread.js:1526]: IIRC it's the lazy clean of the problem we introduced with `@-`mention in chat, right?
If so, would be a great idea to make a task to remove this code in 6 months or so (we can schedule activity to do it).
What do you think @seb-odoo ?

Comment (seb-odoo) [reply addons/mail/static/src/models/thread/thread.js:1526]: We can do it whenever we want in master but it needs to be together with a migration script. It's just not priority right now.

Comment (seb-odoo) [reply addons/mail/static/src/models/thread/thread.js:1526]: Also note that if we ever merge members & follower which is likely to happen this year, that task will already take care of this problem.

Comment (alexkuhn) [reply addons/mail/static/src/models/thread/thread.js:1526]: Ok, I was just suggesting planning removal in the future just so we don't keep weird stuffs indefinitely in code when we shouldn't need it. It's fine as long as we don't forget it.

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**